index.ts

```diff
@@ -176,33 +159,29 @@ function parseUrls(message: Message): string[] {
 async function resolveHandler(message: Message): Promise<HandlerResult> {
     const url = parseUrls(message)[0];
 
-    if (!url) {
-        return Promise.reject("no resource url.");
-    }
-
+    if (!url) throw new Error("no resource url.");
     console.log(`intercepted url '${url}'`);
 
     for (const handler of handlers) {
         try {
             console.log(`trying handler '${handler.id}'`);
-            return await handler.handle(url);
```

Comment:
this await was here in case we wanted a handler fail through to a fallback in case this was ever necessary, but it probably won't ever be.